### PR TITLE
Implemented the Intern Posting backend code.

### PR DESCRIPTION
**Pull request by KushaRa**
Created an Advertisement schema in the models directory to define the structure of advertisement documents in MongoDB database. 
Developed the 'AdvertisementController' to manage advertisement-related operations, such as submitting and fetching advertisements. 
Additionally, set up routes for these operations in the routes directory, linking them to the controller functions. and configured the main application file (index.js) to include these routes under the /api endpoint, ensuring that all advertisement-related requests are appropriately handled by the API.

### CODE REVIEW

**Review by Lakshi99 — COMMENTED** [review]
nice work

File: controllers/advertismentController.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+const Advertisement = require('../models/advertisment');
+
+exports.submitAdvertisement = async (req, res) => {
+  const { company_name, title, duration, job_type, location, email, description } = req.body;
+  const newAdvertisement = new Advertisement({ company_name, title, duration, job_type, location, email, description });
+
+  try {
+    await newAdvertisement.save();
+    res.status(200).send('Post submitted successfully!');
```
Comment (bihanviranga) [inline controllers/advertismentController.js]: Suggestion: I think status code 201 is more suitable here.

File: routes/authRoutes.js [hunk 2]
```diff
@@ -1,6 +1,6 @@
 const express = require("express");
 const router = express.Router();
-const usercontroller = require("../controllers/authController.js");
+const usercontroller = require("../controllers/advertismentController");
```
Comment (bihanviranga) [inline routes/authRoutes.js]: Why is this called `usercontroller` ?

**Review by bihanviranga — COMMENTED** [review]
Great work @KushaRa ! Added some comments.

File: controllers/advertismentController.js (hunk 1 above)
Comment (KushaRa) [reply controllers/advertismentController.js]: Thank you for the suggestion! I'll change the status code to 201, as it better indicates successful resource creation.

File: routes/authRoutes.js (hunk 2 above)
Comment (KushaRa) [reply routes/authRoutes.js]: Thanks for pointing that out! The require statement for 'authController' was incorrectly named 'advertisementController'. I'll change it to 'authController' to reflect the correct controller.

**Pull request merged**